### PR TITLE
fixed control system relative bug

### PR DESCRIPTION
**Pull request by irwineffect**
see #119

### CODE REVIEW

File: src/movement/control_system.cpp [hunk 1]
```diff
@@ -235,26 +235,51 @@ namespace robosub
...
+                    if(control_values[i] == 0)
                     {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -90.0, 90.0);
+                        //if the previous goal was absolute, we can just reuse
+                        //the last absolute goal
+                        if (this->goal_types[i] ==
+                            robosub::control::STATE_ABSOLUTE)
+                        {
+                            this->goals[i] = this->goals[i];
```
Comment (skallaher) [inline src/movement/control_system.cpp]: You're assigning this value to itself... This is unnecessary.

File: src/movement/control_system.cpp [hunk 2]
```diff
@@ -235,26 +235,51 @@ namespace robosub
                      * by adding in the new goal to the current goal to make it
                      * relative to the current state.
                      */
```
Comment (skallaher) [inline src/movement/control_system.cpp:236]: This comment doesn't seem quite as accurate, perhaps refactor it.

**Review by skallaher — CHANGES_REQUESTED** [review]

File: src/movement/control_system.cpp (hunk 2 above)
Comment (irwineffect) [reply src/movement/control_system.cpp:236]: This is still correct, all I've done is update the logic to work as expected for a relative change of zero.

File: src/movement/control_system.cpp (hunk 1 above)
Comment (irwineffect) [reply src/movement/control_system.cpp]: I'm just being explicit, I suppose I could have an empty if statement here with a comment. I'll fix that.

**Review by skallaher — APPROVED** [review]
Looks good!

File: src/movement/control_system.cpp [hunk 3]
```diff
@@ -232,29 +232,56 @@ namespace robosub
...
+                    //if we just want to maintain our current state
+                    if(control_values[i] == 0)
                     {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -90.0, 90.0);
+                        //if the previous goal was absolute, we can just reuse
+                        //the last absolute goal
+                        if (this->goal_types[i] ==
+                            robosub::control::STATE_ABSOLUTE)
+                        {
```
Comment (bkallaher) [inline src/movement/control_system.cpp:245]: Maybe this if else should be switched with no else statement, just a comment about what it would be.

**Review by bkallaher — CHANGES_REQUESTED** [review]
Just a small fix.

**Review by bkallaher — APPROVED** [review]
Looks good.

**Pull request merged**